flow/flow.go

```diff
@@ -312,25 +312,30 @@ func (p *Packet) TransportFlow() (gopacket.Flow, error) {
 // Key returns the unique flow key
 // The unique key is calculated based on parentUUID, network, transport and applicable layers
 func (p *Packet) Key(parentUUID string, opts Opts) string {
-	var uuid uint64
-
+	hasher := xxHash64.New(0)
 	// uses L2 is requested or if there is no network layer
 	if opts.LayerKeyMode == L2KeyMode || p.NetworkLayer() == nil {
 		if layer := p.LinkLayer(); layer != nil {
-			uuid ^= layer.LinkFlow().FastHash()
+			Hash(layer.LinkFlow(), hasher, false)
+			b := make([]byte, 8)
+			binary.BigEndian.PutUint64(b, uint64(linkID(p)))
+			hasher.Write(b)
 		}
 	}
 	if layer := p.NetworkLayer(); layer != nil {
-		uuid ^= layer.NetworkFlow().FastHash()
+		Hash(layer.NetworkFlow(), hasher, false)
+		b := make([]byte, 8)
+		binary.BigEndian.PutUint64(b, uint64(networkID(p)))
+		hasher.Write(b)
+
 	}
-	if tf, err := p.TransportFlow(); err == nil {
-		uuid ^= tf.FastHash()
+	if layer := p.TransportLayer(); layer != nil {
```

Comment:
To follow my comment on the Hash function, previously TransportFlow was used and that were the geneve/vxlan thing was handled. I think you just need to use it instead of layer.TransportFlow so that you should be able to remove the geneve/vxlan thing from your hash function. You will see that the VNI is also used there.

Reply:
you are correct again;)
done